scriptworker/task.py

```diff
@@ -396,21 +400,34 @@ def prepare_to_run_task(context, claim_task):
         'preexec_fn': lambda: os.setsid(),
     }
     context.proc = await asyncio.create_subprocess_exec(*context.config['task_script'], **kwargs)
-    loop.call_later(context.config['task_max_timeout'], max_timeout, context, context.proc, context.config['task_max_timeout'])
 
     tasks = []
+    timeout_exc = None
     with get_log_filehandle(context) as log_filehandle:
-        tasks.append(pipe_to_log(context.proc.stderr, filehandles=[log_filehandle]))
-        tasks.append(pipe_to_log(context.proc.stdout, filehandles=[log_filehandle]))
+        tasks.append(asyncio.ensure_future(
+            pipe_to_log(context.proc.stderr, filehandles=[log_filehandle])
+        ))
+        tasks.append(asyncio.ensure_future(
+            pipe_to_log(context.proc.stdout, filehandles=[log_filehandle])
+        ))
+        timeout_task = asyncio.ensure_future(
+            max_timeout(context, context.proc, context.config['task_max_timeout'])
+        )
+        tasks.append(timeout_task)
         await asyncio.wait(tasks)
         exitcode = await context.proc.wait()
         status_line = "exit code: {}".format(exitcode)
         if exitcode == -11:
```

Comment:
This probably should be `exitcode < 0`.

Reply:
Well, all negative exit codes represent the process exiting with a signal, so we should log all of them like that. We may also want to do interesting stuff with things in `reverse_statues`.

Reply:
Sure. This was in reaction to a specific issue where we were hitting -11 and wanting that in the logs, but I don't see any harm with including all negative statuses.